File: lab7_1/lexer_ts/src/Position.ts
class Position {
    private line: number
    private column: number;

    constructor(line: number, column: number) {
        this.line = line;
        this.column = column;
    }

    public getLine(): number {
        return this.line;
    }

    public getColumn(): number {
        return this.column
    }

    public setLine(lineNumber: number): void {
        this.line = lineNumber;
    }

    public setNextLine(): void {
        this.line++;
    }

    public setColumn(columnNumber: number): void {
        this.column = columnNumber;
    }

    public setNextColumn(): void {
        this.column++;
    } 

    public toString(): string {
        return '(' + this.line + ',' + this.column + ')'; 
    }

}

export { Position }

File: lab4/Position.java
package lab4;

public class Position {
    private int line, column;

    public Position(int line, int column) {
        this.line = line;
        this.column = column;
    }

    public Position(Position currentPos) {
        this.line  = currentPos.getLine();
        this.column   = currentPos.getColumn();
    }

    public int getLine() {
        return this.line;
    }

    public void setLine(int lineNumber) {
        this.line = lineNumber;
    }

    public void setNextLine() {
        this.line++;
    }

    public int getColumn() {
        return column;
    }

    public void setNextColumn() {
        this.column++;
    }

    public void setColumn(int columnNumber) {
        this.column = columnNumber;
    }

    @Override
    public String toString() {
        return "(" + line + "," + column + ")";
    }

}